src/main/java/com/philips/research/bombase/core/nuget/domain/NugetAPI.java

```diff
@@ -0,0 +1,120 @@
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+
+import java.net.URI;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public interface NugetAPI {
+    // TODO Is this a proper value?
+    int NUGET_SCORE = 80;
+
+    @GET("registration5-semver1/{project}/{version}.json")
+    Call<CatalogResponseJson> getCatalogEntry(@Path("project") String project,
+                                              @Path("version") String version);
+    @GET("{catalogEntryUrl}")
+    Call<ResponseJson> getDefinition(@Path("catalogEntryUrl") String catalogEntryUrl);
+
+    @SuppressWarnings("NotNullFieldNotInitialized")
+    class CatalogResponseJson {
+        @JsonProperty("@id")
+        public String id;
+        @JsonProperty("@type")
+        public List<String> type;
+        @JsonProperty("catalogEntry")
+        public String catalogEntry;
+        public boolean listed;
+        public Date published;
+        public String registration;
+
+    }
+    @SuppressWarnings("NotNullFieldNotInitialized")
+    class ResponseJson implements PackageMetadata {
+        @JsonProperty("title")
+        @NullOr String name;
+        @NullOr String description;
+        @JsonProperty("projectUrl")
+        @NullOr URI homepage;
+        @JsonProperty("licenseExpression")
+        @NullOr String license;
+        @JsonProperty("authors")
+        @NullOr String author;
+        @NullOr JsonNode repository;
+        @JsonProperty("packageHash")
+        @NullOr String packageHash;
+
+        @Override
+        public int score(Field field) {
+            return NUGET_SCORE;
+        }
+
+        @Override
+        public Optional<String> getTitle() {
+            return Optional.ofNullable(name);
+        }
+
+        @Override
+        public Optional<String> getDescription() {
+            return Optional.ofNullable(description);
+        }
+
+        @Override
+        public Optional<List<String>> getAuthors() {
+            if (author == null) {
+                return Optional.empty();
+            }
+            return Optional.of(Stream.of(author.split(","))
+                    .map(String::stripLeading)
+                    .collect(Collectors.toList()));
+        }
+
+        @Override
+        public Optional<URI> getHomepage() {
+            return Optional.ofNullable(homepage);
+        }
+
+        @Override
+        public Optional<String> getDeclaredLicense() {
+            return Optional.ofNullable(license);
+        }
+
+        @Override
+        public Optional<String> getSourceLocation() {
+            if (repository == null) {
+                return Optional.empty();
+            }
+            if (repository.isObject()) {
+                return Optional.ofNullable(repository.get("url").textValue());
+            }
+            return Optional.of(repository.textValue());
```

Comment:
Are you sure that the response (like the NPMJS repo) sometimes reports a plain string and for other packages an object with an `"url"` property?

If not, then it might be better to turn the object into a `static class` with an `url` field. (This makes the code a bit easier to read.)